src/pt/leituramanga/src/eu/kanade/tachiyomi/extension/pt/leituramanga/LeituraManga.kt

```diff
@@ -0,0 +1,133 @@
+        return MangasPage(mangas, hasNext)
+    }
+    // ================= Search ==================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/api/manga/".toHttpUrl().newBuilder()
+            .addQueryParameter("limit", "24")
+            .addQueryParameter("page", page.toString())
+
+        if (query.isNotEmpty()) {
+            url.addQueryParameter("keyword", query)
+        }
+
+        filters.forEach { filter ->
+            when (filter) {
+                is GenreFilter -> {
+                    if (filter.selectedValue.isNotEmpty()) {
+                        url.addQueryParameter("genre", filter.selectedValue)
+                    }
+                }
+                is StatusFilter -> {
+                    if (filter.selectedValue.isNotEmpty()) {
+                        url.addQueryParameter("status", filter.selectedValue)
+                    }
+                }
+                is SortFilter -> {
+                    url.addQueryParameter("sort", filter.selectedValue)
+                }
+                else -> {}
+            }
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun searchMangaParse(response: Response) = latestUpdatesParse(response)
+
+    override fun getFilterList() = FilterList(
+        GenreFilter(),
+        StatusFilter(),
+        SortFilter(),
+    )
+    // ================= Details ==================
+
+    override fun mangaDetailsRequest(manga: SManga) =
+        GET("$apiUrl/api/manga/slug/${manga.url.removeSuffix("/").substringAfterLast("/")}", headers)
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val result = response.parseAs<MangaResponseDto<MangaDto>>()
+        return result.data.toSManga(cdnUrl)
+    }
+
+    override fun getMangaUrl(manga: SManga) = baseUrl + manga.url
+    // ================= Chapters ==================
+
+    override fun chapterListRequest(manga: SManga) = mangaDetailsRequest(manga)
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val mangaDto = response.parseAs<MangaResponseDto<MangaDto>>().data
+        val request = GET("$apiUrl/api/chapter/get-by-manga-id?mangaId=${mangaDto._id}&page=1&limit=9999", headers)
+        val result = client.newCall(request).execute().parseAs<MangaResponseDto<ChapterListDto>>()
+
+        return result.data.data.map { it.toSChapter(mangaDto.slug) }
+    }
+
+    override fun getChapterUrl(chapter: SChapter) = baseUrl + chapter.url
+    // ================= Pages ==================
+
+    override fun pageListParse(response: Response): List<Page> {
+        val document = response.asJsoup()
+
+        return document.select("div[data-index] img").mapIndexed { i, element ->
+            Page(i, "", element.attr("abs:src"))
+        }
+    }
+
+    override fun imageUrlParse(response: Response): String = throw Exception("Not used")
```

Comment:
```suggestion
    override fun imageUrlParse(response: Response): String = throw UnsupportedOperationException()
```

per the contributing guidelines.